pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -98,6 +98,10 @@ public abstract class FieldSpec implements Comparable<FieldSpec>, Serializable {
     }
   }
 
+  public enum MaxLengthExceedStrategy {
+    TRIM_LENGTH, FAIL_INGESTION, SUBSTITUTE_DEFAULT_VALUE
```

Comment:
Let's add a strategy to do nothing when the max length is exceeded. The reason being currently we don't do anything for BYTES and JSON when the value length exceeds the max length, but in the future we also want to limit their length. It can also be useful when user want to turn off the max length limit.

Reply:
Added a `NO_ACTION` enum to put values as is.